clang/lib/Driver/ToolChains/Gnu.cpp

```diff
@@ -2477,7 +2477,7 @@ void Generic_GCC::GCCInstallationDetector::AddDefaultGCCPrefixes(
       "x86_64-redhat-linux",    "x86_64-suse-linux",
       "x86_64-manbo-linux-gnu", "x86_64-linux-gnu",
       "x86_64-slackware-linux", "x86_64-unknown-linux",
-      "x86_64-amazon-linux"};
+      "x86_64-amazon-linux",    "x86_64-gnu"};
```

Comment:
There is a comment above that we should not add new elements to these arrays.

Is it here so that a target triple of `x86_64-pc-hurd-gnu` can find a GCC installation named `x86_64-gnu` for Debian? If that's the case, use code to indicate the special case for Hurd so that the logic will not affect other x86_64 OS triples.

Reply:
That was it indeed, I have replaced with some code.

It's not debian that picked it up, but the GNU toolchains (GNU/Hurd was supposed to be the "pure GNU" system).